### PR TITLE
Make DOMTokenList a live list and fix the class parsing

### LINKED ISSUE #24: DOMTokenList shouldn't contain empty tokens
Per the current implementation, `DOMTokenList` includes an empty token if the `class` attribute starts or ends with space, which [isn't the](https://dom.spec.whatwg.org/#ref-for-concept-element-attributes-change-ext%E2%91%A3) [spec-mandated behavior](https://infra.spec.whatwg.org/#split-on-ascii-whitespace).

Additionally, Infra's definition of ASCII whitespace is the very limited `[\t\n\f\r ]`, rather than `\s` which includes all sort of Unicode space characters.

### PR DESCRIPTION
**Pull request by andreubotella**
Closes #24.

### CODE REVIEW

File: src/dom/element.ts [hunk 1]
```diff
@@ -4,9 +4,52 @@ import { Node, NodeType, Text } from "./node.ts";
 import { NodeList, nodeListMutatorSym } from "./node-list.ts";
 
 export class DOMTokenList extends Set<string> {
+  #onChange: (cls: string) => void;
```
Comment (b-fuze) [inline src/dom/element.ts:7]: Please use `className` or `classNames` instead of `cls`

File: src/dom/element.ts [hunk 2]
```diff
@@ -4,9 +4,52 @@ import { Node, NodeType, Text } from "./node.ts";
...
+    this.#onChange([...this].join(" "));
+    return this;
+  }
+
+  clear() {
+    super.clear();
+    this.#onChange("");
+  }
+
+  delete(token: string): boolean {
```
Comment (b-fuze) [inline src/dom/element.ts:28]: While we're at it, `delete` should be a private API with a public `remove` method

File: src/dom/element.ts [hunk 3]
```diff
@@ -220,15 +267,21 @@ export class Element extends Node {
   }
 
   setAttribute(name: string, value: any) {
-    this.attributes[name] = "" + value;
+    const strValue = String(value);
```
Comment (b-fuze) [inline src/dom/element.ts:270]: Why are you using the `String()` function instead of `"" + value` here?

File: src/dom/element.ts [hunk 4]
```diff
@@ -220,15 +267,21 @@ export class Element extends Node {
...
-      this.#currentId = value;
+      this.#currentId = strValue;
+    } else if (name === "class") {
+      this.classList._update(strValue);
     }
   }
 
   removeAttribute(name: string) {
-    this.attributes[name] = null as any as string;
+    this.attributes[name] = (null as any) as string;
```
Comment (b-fuze) [inline src/dom/element.ts:281]: Why did you put parens on `(null as any)`?

**Review by b-fuze — CHANGES_REQUESTED** [review]
Thank you for the PR, please check my comments

File: src/dom/element.ts (hunk 4 above)
Comment (andreubotella) [reply src/dom/element.ts:281]: That was my IDE's formatting, I removed the changes elsewhere but missed it here.

File: src/dom/element.ts (hunk 3 above)
Comment (andreubotella) [reply src/dom/element.ts:270]: I find the `String()`, `Number()`, etc. functions more clear than forcing an implicit typecast. But I'll change it back if you prefer.

File: src/dom/element.ts (hunk 4 above)
Comment (b-fuze) [reply src/dom/element.ts:281]: Yeah, change it to the coercion thanks

File: src/dom/element.ts (hunk 3 above)
Comment (b-fuze) [reply src/dom/element.ts:270]: Yeah, change it to the coercion thanks

**Comment by b-fuze** [comment]
LGTM, thanks

**Pull request merged**